draft-ietf-tls-tls13.md

```diff
@@ -2969,24 +2968,33 @@ MUST generate a fatal "unexpected_message" alert.  After sending a
 KeyUpdate message, the sender SHALL send all its traffic using the
 next generation of keys, computed as described in
 {{updating-traffic-keys}}. Upon receiving a KeyUpdate, the receiver
-MUST update their receiving keys and if they have not already updated
-their sending state up to or past the then current receiving
-generation MUST send their own KeyUpdate prior to sending any other
-messages.  This mechanism allows either side to force an update to the
-entire connection. Note that implementations may receive an arbitrary
+MUST update its receiving keys.
+
+Each implementation MUST receive a min_send_generation counter that
+is initialized to 0 at the time the traffic keys are installed.
+Upon receiving a KeyUpdate, if the receiver has sent any data records since receiving the
+last KeyUpdate it MUST also increment its min_send_generation value
+by 1. Otherwise, the min_send_generation MUST remain unchanged. Prior
+to sending a record, if min_send_generation is greater than the
+current sending generation, the sender MUST first send a KeyUpdate.
```

Comment:
After chatting out-of-band, I misunderstood this scheme and it doesn't cause ping-pong. But instead it doesn't let you ask the peer to KeyUpdate if, say, the peer sent 1,000 KeyUpdates in a row two hours ago because they thought it was funny. min_send_generation will be far behind send_generation.